pkg/virt-controller/services/template.go

```diff
@@ -150,6 +193,22 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 	resources := k8sv1.ResourceRequirements{}
 	vmiResources := vmi.Spec.Domain.Resources
 
+	isMemoryFiledExist := func(resource k8sv1.ResourceList) bool {
+		_, ok := resource[k8sv1.ResourceMemory]
+		return ok
+	}
+
+	// Copy namespace memory limits (if exists) to the VM spec
+	if vmiResources.Limits == nil || !isMemoryFiledExist(vmiResources.Limits) {
```

Comment:
Can you move the whole `vmi` manipulation logic to the preset controller? This is the location where all this magic should happen. The preset controller will then in the future be moved to the modifying webhook once introduce it.